File: src/Domain/Entities/Project.cs
using StartupConnect.Domain.Enums;

namespace StartupConnect.Domain.Entities;

public sealed class Project : BaseEntity
{
    public Guid OwnerUserId { get; set; }

    public User OwnerUser { get; set; } = null!;

    public string Title { get; set; } = string.Empty;

    public string Slug { get; set; } = string.Empty;

    public string Summary { get; set; } = string.Empty;

    public string Problem { get; set; } = string.Empty;

    public string Solution { get; set; } = string.Empty;

    public string? TargetMarket { get; set; }

    public string? BusinessModel { get; set; }

    public string? FundingNeeds { get; set; }

    public string? PitchDeckUrl { get; set; }

    public ProjectStatus Status { get; set; } = ProjectStatus.Draft;

    public ProjectStage Stage { get; set; } = ProjectStage.Idea;

    public bool IsRecruiting { get; set; } = true;

    public bool IsDeleted { get; set; }

    public DateTimeOffset? SubmittedAt { get; set; }

    public ICollection<ProjectMember> Members { get; set; } = [];
}

File: frontend/src/types/project.ts
import { ProjectStatus, ProjectStage, ProjectVisibility } from './enums';

export interface ProjectSummaryDto {
  id: string;
  title: string;
  slug: string;
  summary: string;
  status: ProjectStatus;
  stage: ProjectStage;
  visibility: ProjectVisibility;
  isRecruiting: boolean;
  createdAt: string;
}

export interface ProjectDetailDto {
  id: string;
  ownerUserId: string;
  title: string;
  slug: string;
  summary: string;
  problem: string;
  solution: string;
  targetMarket?: string;
  businessModel?: string;
  fundingNeeds?: string;
  pitchDeckUrl?: string;
  status: ProjectStatus;
  stage: ProjectStage;
  isRecruiting: boolean;
  visibility: ProjectVisibility;
  requiresNda: boolean;
  requiredRoles: ProjectRequiredRoleDto[];
  requiredSkills: ProjectSkillDto[];
  createdAt: string;
  updatedAt?: string;
}

export interface CreateProjectDraftRequest {
  title: string;
  summary: string;
  problem: string;
  solution: string;
  stage: ProjectStage;
  visibility: ProjectVisibility;
}

export interface UpdateProjectRequest {
  title: string;
  summary: string;
  problem: string;
  solution: string;
  targetMarket?: string;
  businessModel?: string;
  fundingNeeds?: string;
  pitchDeckUrl?: string;
  stage: ProjectStage;
  visibility: ProjectVisibility;
  isRecruiting: boolean;
  requiredRoles: UpsertProjectRequiredRoleDto[];
  requiredSkillIds: string[];
}

export interface UpsertProjectRequiredRoleDto {
  roleName: string;
  description?: string;
  slots: number;
  isOpen: boolean;
}

export interface ProjectRequiredRoleDto {
  id: string;
  roleName: string;
  description?: string;
  slots: number;
  isOpen: boolean;
}

export interface ProjectSkillDto {
  id: string;
  name: string;
}

export interface ProjectVersionDto {
  id: string;
  versionNumber: number;
  changeReason: string;
  createdAt: string;
}

export interface LandingProject {
  id: string;
  title: string;
  summary: string;
  field: string;
  stage: string;
  status: string;
  roles: string[];
  memberCount: number;
  seekingInvestor: boolean;
  href: string;
}
